src/realm/object-store/results.cpp

```diff
@@ -536,6 +536,10 @@ size_t Results::index_of(T const& value)
             if (!mixed.is_null()) {
                 val = mixed.get<U>();
             }
+            else {
+                if (mixed == value)
```

Comment:
Note that `Mixed::operator==()` uses the a loose comparison, so `double(123.0) == int(123)`. Do you want that here? If not, your can check `Mixed::get_type()` as well.

Reply:
I don't think the loose comparison is an issue but how would you compare `Mixed::get_type()` to `T const&`?

Reply:
It's a bit hidden if you don't know about it, but there are helpers to get the data type from a C++ type: `ColumnTypeTraits<T>::id`